packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js

```diff
@@ -0,0 +1,201 @@
+          is: ['eq', value],
+          isAfter: ['gt', value],
+          isBefore: ['lt', value],
+          isNot: ['nq', value],
+          smaller: ['lt', value],
+          startsWith: ['like', `${value}%`],
+        }[operator],
+      );
+    });
+
+    this.isLoadingAccesses = true;
+    this.hideAccessForm();
+
+    return accessesQuery
+      .execute(null, true)
+      .$promise.catch(({ data: error }) => {
+        this.alertError(error);
+        return { data: [] };
+      })
+      .then(({ data }) => {
+        this.accesses = data;
+        if (!this.accesses.length) {
+          return this.showAccessForm().then(() => this.accesses);
+        }
+        return this.accesses;
+      })
+      .then((data) => ({
+        data,
+        meta: { totalCount: data.length },
+      }))
+      .finally(() => {
+        this.isLoadingAccesses = false;
+      });
+  }
+
+  showAccessForm() {
+    let promise = this.$q.when();
+
+    if (!this.authorizedAccesses) {
+      const { partitionApiUrl } = this;
+
+      this.isLoadingAccessForm = true;
+
+      promise = this.$q
+        .all({
+          ips: this.$http.get(`${partitionApiUrl}/authorizableIps`),
+          blocks: this.$http.get(`${partitionApiUrl}/authorizableBlocks`),
+        })
+        .then(({ ips: { data: ips }, blocks: { data: blocks } }) => {
+          this.authorizedAccesses = [
+            ...ips
+              .filter((x) => !this.accesses.find(({ ip }) => ip === `${x}/32`))
+              .map((ip) => ({ type: 'IP', ip })),
+            ...blocks
+              .filter((block) => !this.accesses.find(({ ip }) => ip === block))
+              .map((block) => ({ type: 'IP/Block', ip: block })),
+          ].sort(
+            ({ ip: ipBlockA }, { ip: ipBlockB }) =>
+              Number(
+                ipBlockA
+                  .replace('/', '.')
+                  .split('.')
+                  .map((n) => n.padStart(3, 0))
+                  .join(''),
+              ) -
+              Number(
+                ipBlockB
+                  .replace('/', '.')
+                  .split('.')
+                  .map((n) => n.padStart(3, 0))
+                  .join(''),
+              ),
+          );
+        })
+        .catch(this.alertError)
+        .finally(() => {
+          this.isLoadingAccessForm = false;
+        });
+    }
+
+    return promise.then(() => {
+      const access = { ip: null, type: null };
+      this.accesses.unshift({ ...access, isForm: true });
+      this.isAccessFormShown = true;
+      this.model = access;
+    });
+  }
+
+  hideAccessForm() {
+    const [access] = this.accesses;
+    if (access?.isForm) {
+      this.accesses.shift();
+      this.isAccessFormShown = false;
+    }
+  }
+
+  createAccess() {
+    const { partitionName } = this.partition;
+    const {
+      ip: { ip },
+      type: { value: type },
+    } = this.model;
+
+    this.isCreatingAccess = true;
+    // this.translate('access_created')
+
+    this.$http
```

Comment:
```suggestion
   return  this.$http
```
It is a good practice to return promises

Reply:
:+1: